src/v/cloud_storage/async_manifest_view.h

```diff
@@ -308,6 +320,29 @@ class async_manifest_view_cursor {
     ss::timer<ss::lowres_clock> _timer;
     const model::offset _begin;
     const model::offset _end;
+    std::optional<model::offset> _stm_start_offset{std::nullopt};
 };
 
+/// Invoke functor with every segment_meta accessible by the cursor
+/// The functor is expected to return stop_iteration::yes if the processing
+/// is completed.
+template<class Fn>
+ss::future<>
+for_each_segment(std::unique_ptr<async_manifest_view_cursor> cursor, Fn func) {
```

Comment:
Just making sure, is this meant to be used anywhere, or just in tests?

Reply:
it's supposed to be used in the followup to unify retention between STM and spillover